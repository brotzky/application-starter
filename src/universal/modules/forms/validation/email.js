import validator from 'validator';

export default function(value) {
  return validator.isEmail(value)
    ? undefined
    : {
        defaultMessage: 'must be a valid email address',
        id: 'form.errors.email',
        values: {},
      };
}
