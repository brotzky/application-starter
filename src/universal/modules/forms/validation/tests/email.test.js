import email from '../email';

describe('[Form Validation] email', () => {
  describe('email()', () => {
    it('should return no errors if a valid email is passed in', () => {
      expect(email('test@test.com')).toMatchSnapshot();
    });

    it('should return an error object when an invalid email is passed in', () => {
      expect(email('bademail')).toMatchSnapshot();
    });
  });
});
