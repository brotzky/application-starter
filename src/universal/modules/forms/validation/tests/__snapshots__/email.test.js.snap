// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Form Validation] email email() should return an error object when an invalid email is passed in 1`] = `
Object {
  "defaultMessage": "must be a valid email address",
  "id": "form.errors.email",
  "values": Object {},
}
`;

exports[`[Form Validation] email email() should return no errors if a valid email is passed in 1`] = `undefined`;
